Test/ApiFunctional/Model/Button/CartTest.php

```diff
@@ -0,0 +1,179 @@
+                'operation' => 'quoteGuestCartItemRepositoryV1Save',
+            ],
+        ];
+
+        $requestData = [
+            'cartItem' => [
+                'quote_id' => $cartId,
+                'sku' => 'amwal_simple',
+                'qty' => 1
+            ]
+        ];
+        $item = $this->_webApiCall($serviceInfoForAddingProduct, $requestData);
+        $this->assertNotEmpty($item);
+
+
+        /** @var Quote $quote */
+        $quote = $this->objectManager->create(Quote::class);
+        $quote->load($item['quote_id']);
+
+        $this->assertNotEmpty($quote->getId());
+
+        $serviceInfoForGetCartButtonConfig = [
+            'rest' => [
+                'resourcePath' => self::RESOURCE_PATH,
+                'httpMethod' => Request::HTTP_METHOD_POST,
+            ],
+            'soap' => [
+                'service' => self::SERVICE_NAME,
+                'serviceVersion' => self::SERVICE_VERSION,
+                'operation' => self::SERVICE_NAME . 'GetCartButtonConfig',
+            ],
+        ];
+        $requestData = [
+            'refIdData' => [
+                'identifier' => '100',
+                'customer_id' => '0',
+                'timestamp' => '1707916143'
+            ],
+            'triggerContext' => 'product-detail-page',
+            'locale' => 'en',
+            'cart_id' => $cartId,
+        ];
+
+        $response = $this->_webApiCall($serviceInfoForGetCartButtonConfig, $requestData);
+        $this->assertIsArray($response);
+
+        // Perform assertions
+        $this->assertArrayHasKey('merchant_id', $response);
+        $this->assertArrayHasKey('amount', $response);
+        $this->assertArrayHasKey('country_code', $response);
+        $this->assertArrayHasKey('dark_mode', $response);
+        $this->assertArrayHasKey('email_required', $response);
+        $this->assertArrayHasKey('address_required', $response);
+        $this->assertArrayHasKey('address_handshake', $response);
+        $this->assertArrayHasKey('ref_id', $response);
+        $this->assertArrayHasKey('label', $response);
+        $this->assertArrayHasKey('disabled', $response);
+        $this->assertArrayHasKey('show_payment_brands', $response);
+        $this->assertArrayHasKey('enable_pre_checkout_trigger', $response);
+        $this->assertArrayHasKey('enable_pre_pay_trigger', $response);
+        $this->assertArrayHasKey('id', $response);
+        $this->assertArrayHasKey('test_environment', $response);
+        $this->assertArrayHasKey('allowed_address_countries', $response);
+        $this->assertArrayHasKey('allowed_address_states', $response);
+        $this->assertArrayHasKey('plugin_version', $response);
+        $this->assertArrayHasKey('post_code_optional_countries', $response);
+        $this->assertArrayHasKey('installment_options_url', $response);
+        $this->assertArrayHasKey('show_discount_ribbon', $response);
+        $this->assertArrayHasKey('discount', $response);
```

Comment:
Maybe we can simplify this by introducing a `const` which contains all the keys that should be present, and then checking these with a foreach?